tools/new_minor_OCP_version.py

```diff
@@ -17,232 +24,452 @@
+        return
+
+    branch, openshift_versions_json = update_ai_repo_to_new_ocp_version(args,
+                                                                        current_assisted_service_ocp_version,
+                                                                        latest_ocp_version, task)
+    logging.info(f"Using versions JSON {openshift_versions_json}")
+    pr = open_pr(args, current_assisted_service_ocp_version, latest_ocp_version, task)
+    test_success = test_changes(args, branch, pr)
+
+    if test_success:
+        fork = create_app_interface_fork(args)
+        app_interface_branch = update_ai_app_sre_repo_to_new_ocp_version(fork, args, latest_ocp_version,
+                                                                         openshift_versions_json, task)
+        pr = open_app_interface_pr(fork, app_interface_branch, current_assisted_service_ocp_version,
+                                    latest_ocp_version,
+                                    task)
+
+        jira_client.add_comment(task, f"Created a PR in app-interface GitLab {pr.web_url}")
+
+
+def test_changes(args, branch, pr):
+    succeeded, url = test_test_infra_passes(args, branch)
+    if succeeded:
         logging.info("Test-infra test passed, removing hold branch")
-        remove_hold_lable(pr)
-        pr.create_issue_comment("test-infra test passed, HOLD label removed, see {}".format(test_infra_result[1]))
+        remove_hold_label(pr)
+        pr.create_issue_comment(f"test-infra test passed, HOLD label removed, see {url}")
     else:
-        logging.WARN("Test-infra test failed, not removing hold label")
-        pr.create_issue_comment("test-infra test failed, HOLD label is set, see {}".format(test_infra_result[1]))
+        logging.warning("Test-infra test failed, not removing hold label")
+        pr.create_issue_comment(f"test-infra test failed, HOLD label is set, see {url}")
 
+    return succeeded
 
-def test_test_infra_passes(branch):
-    jkusername, jkpassword = get_login(args.jenkins_user_password)
-    j = jenkins.Jenkins(JENKINS_URL, username=jkusername, password=jkpassword)
-    next_build_number = j.get_job_info(TEST_INFRA_JOB)['nextBuildNumber']
-    j.build_job(TEST_INFRA_JOB, parameters={"SERVICE_BRANCH": branch, "NOTIFY":False, "JOB_NAME": "Update_ocp_version"})
+
+def test_test_infra_passes(args, branch):
+    username, password = get_login(args.jenkins_user_password)
+    jenkins_client = jenkins.Jenkins(JENKINS_URL, username=username, password=password)
+    next_build_number = jenkins_client.get_job_info(TEST_INFRA_JOB)['nextBuildNumber']
+    jenkins_client.build_job(TEST_INFRA_JOB,
+                             parameters={"SERVICE_BRANCH": branch, "NOTIFY": False, "JOB_NAME": "Update_ocp_version"})
     time.sleep(10)
-    while not j.get_build_info(TEST_INFRA_JOB, next_build_number)["result"]:
+    while not jenkins_client.get_build_info(TEST_INFRA_JOB, next_build_number)["result"]:
```

Comment:
Consider extracting this to `wait_for_job_to_finish` function

Reply:
https://github.com/eranco74/assisted-installer-deployment/pull/41/files#r571624504